app/src/main/java/com/example/moviedb/data/repository/GenreRepository.kt

```diff
@@ -1,19 +1,24 @@
 package com.example.moviedb.data.repository
 
-import com.example.moviedb.data.ApiService
+import com.example.moviedb.data.service.api.ApiService
 import com.example.moviedb.data.model.Genre
+import com.example.moviedb.data.service.GenreResponse
 import io.reactivex.Single
 
 class GenreRepository(private val service: ApiService) {
 
-    companion object{
-        lateinit var genreRepository: GenreRepository
+    companion object {
+        var genreRepository: GenreRepository? = null
 
-        fun getInstance(service: ApiService): GenreRepository{
-            return GenreRepository(service)
+        fun getInstance(service: ApiService): GenreRepository? {
```

Comment:
nên viết ntn nè: 
```suggestion
        var instance: GenreRepository ? = null 
        fun getInstance(service: ApiService): GenreRepository = 
             instance ?: GenreRepository(service).also { instance = it }
```
có nghĩa: nếu không null thì trả về instance , còn không thì trả về giá trị đằng sau, đồng thời gán luôn instance = it
Nếu hàm chỉ là một biểu thức trả về thì sử dụng dấu = thay vì { return } theo convention 
https://kotlinlang.org/docs/reference/coding-conventions.html#function-formatting